### PR TITLE
Update documentation of `DataFormatVecPermute` operation

### LINKED ISSUE #53157: Outdated documentation for `DataFormatVecPermute`
This came up while creating a TF-TRT converter for `DataFormatVecPermute` in #52942 
cc @bixia1 

## URL(s) with the issue:

 - Python: https://www.tensorflow.org/api_docs/python/tf/raw_ops/DataFormatVecPermute
 - C++: https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/data-format-vec-permute
 - Java: https://www.tensorflow.org/jvm/api_docs/java/org/tensorflow/op/nn/DataFormatVecPermute

## Description of the issue (what needs changing):

- Examples imply that an input of shape `(2, 4)` is valid, but that should be `(4, 2)`.
- The format strings can be of size 4 or 5 (e.g `"NHWC"`, `"NDHWC"`).
- The first dimension of the input shape vector can be `src_format.size()` or `src_format.size() - 2`, in which case it is assumed that non-spatial dimensions are omitted.

As an example, here is a valid Python code that is not covered by the documentation:

```python
import tensorflow as tf
a = tf.constant([1, 2, 3, 4, 5])
print(tf.raw_ops.DataFormatVecPermute(x=a, src_format="NDHWC", dst_format="NCDHW"))
```
Output:
```
tf.Tensor([1 5 2 3 4], shape=(5,), dtype=int32)
```

## Submit a pull request?

I'm planning to submit a PR.

### PR DESCRIPTION
**Pull request by Nyrio**
Resolves #53157 
cc @bixia1 

- Fixes invalid shape in the example.
- Updated description of supported inputs.

### CODE REVIEW

**Comment by gbaned** [comment]
@rohan100jain  Can you please review this PR ? Thank you!

File: tensorflow/core/api_def/base_api/api_def_DataFormatVecPermute.pbtxt [hunk 1]
```diff
@@ -26,25 +26,36 @@ END
   }
   summary: "Permute input tensor from `src_format` to `dst_format`."
   description: <<END
-Input tensor must be a vector of size 4, or a 4x2 tensor.
+Given a source and destination format strings of length n=4 or 5, the input
```
[the remark this answers is not in the document]
Comment (bixia1) [reply tensorflow/core/api_def/base_api/api_def_DataFormatVecPermute.pbtxt:29]: "a source and destination format strings" doesn't look right to me. Would "source and destination format strings" work?

**Comment by bixia1** [comment]
Do you want to update wrappers.go and tf_generated_ops.td and then the PR description here?

**Comment by Nyrio** [comment]
> Once this updated documentation is reviewed, should I generate wrappers.go and tf_generated_ops.td?

@bixia I meant this as a question, not a todo item. It seems to me that:

- `wrappers.go`, according to [its  history](https://github.com/tensorflow/tensorflow/commits/master/tensorflow/go/op/wrappers.go), is updated by an automated job every week or so.
- `tf_generated_ops.td` might be generated by a script that isn't public? I didn't find any build target or script that updated the file and [this StackOverflow answer](https://stackoverflow.com/a/61274876/8259873) suggests there is such a script but it isn't open source.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**